src/data/localization.rs

```diff
@@ -397,34 +390,58 @@ impl Localization {
         }
     }
 
-    pub fn mark_used(&self, key: &str) {
-        for lang in self.iter_lang_idx() {
-            if let Some(entry) = self.locas[lang].get(key) {
-                entry.used.store(true, Relaxed);
+    /// Marks a localization key as used for all languages.
+    /// Returns whether the key exists for any language (same as [`Localization::exists`]).
+    pub fn mark_used(&self, key: &str) -> bool {
+        let mut exists = false;
+        for lang in self.iter_lang() {
+            exists |= self.mark_used_lang(key, lang);
+        }
+        exists
+    }
+
+    /// Marks a localization key as used for all languages.
+    /// Returns a [`Vec<&str>`] containing the languages for which the key does not exist.
+    fn mark_used_missing(&self, key: &str) -> Vec<&'static str> {
```

Comment:
I think `mark_used_return_missing` would be a better name for this Frankenstein function :)